.github/workflows/esp32.yml

```diff
@@ -0,0 +1,77 @@
+name: ESP32
+on:
+  push:
+    branches: [master]
+    paths:
+      - '.github/workflows/esp32.yml'
+      - 'CMakeLists.txt'
+      - 'cmake/**'
+      - 'src/*'
+      - 'src/layer/*'
+      - 'src/layer/x86/**'
+      - 'src/layer/vulkan/**'
+      - 'tests/**'
+  pull_request:
+    branches: [master]
+    paths:
+      - '.github/workflows/esp32.yml'
+      - 'CMakeLists.txt'
+      - 'cmake/**'
+      - 'src/*'
+      - 'src/layer/*'
+      - 'src/layer/x86/**'
+      - 'src/layer/vulkan/**'
+      - 'tests/**'
+
+concurrency:
+  group: esp32-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: ESP32
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y cmake ninja-build ccache
+
+      - name: Cache-ESP-IDF
+        id: Cache-ESP-IDF
+        uses: actions/cache@v2
+        with:
+          path: esp-idf-install
+          key: esp-idf-${{ runner.os }}-install
+
+      - name: Clone and Install ESP-IDF
+        if: steps.Cache-ESP-IDF.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/espressif/esp-idf
+          cd esp-idf
+          git submodule update --init --recursive
+          ./install.sh
```

Comment:
why do we cache the idf for dev branch ?
use action/checkout for this job, with idf release version but not the dev branch